File: sort_associative_array_by_key_descend/sort_associative_array_by_key_descend.pl
use strict;
my $example_array = {1 => 'January', 12 => 'December', 3 => 'March' };
foreach my $key (reverse sort {$a <=> $b} keys %{$example_array}) {
   print $key. " => " . $example_array->{$key} ."\n";
}

File: sort_associative_array_by_key_descend/sort_associative_array_by_key_descend.js
const example_array = {1:"January", 12:"December", 3:"March"};
var keys=[];
for(var key in example_array) {
  keys.push(key);
}
keys.sort((a,b)=>{return b-a;});
for(var key of keys) {
  console.log(key + " => " + example_array[key]);
}